src/main/content/game/pool-cdn.ts

```diff
@@ -0,0 +1,103 @@
+import { CONTENT_PATH } from "@main/config/app";
+import * as fs from "fs";
+import path from "path";
+import { DownloaderHelper } from "node-downloader-helper";
+import { logger } from "@main/utils/logger";
+
+import type { DownloadInfo } from "@main/content/downloads";
+import { Downloader } from "@main/content/abstract-content";
+import { removeFromArray } from "$/jaz-ts-utils/object";
+import { GameContentAPI } from "@main/content/game/game-content";
+import { extract7z } from "@main/utils/extract-7z";
+
+const log = logger("pool-cdn.ts");
+
+export class PoolCdnDownloader extends Downloader {
+    cdnUrl = "https://pool-init.beyondallreason.dev";
+    poolDataUrl = `${this.cdnUrl}/data.7z`;
+
+    constructor(gameApi: GameContentAPI) {
+        super();
+        this.onDownloadStart.add((downloadInfo) => {
+            gameApi.onDownloadStart.dispatch(downloadInfo);
+        });
+        this.onDownloadComplete.add((downloadInfo) => {
+            gameApi.onDownloadComplete.dispatch(downloadInfo);
+        });
+        this.onDownloadProgress.add((downloadInfo) => {
+            gameApi.onDownloadProgress.dispatch(downloadInfo);
+        });
+        this.onDownloadFail.add((downloadInfo) => {
+            gameApi.onDownloadFail.dispatch(downloadInfo);
+        });
+    }
+
+    /**
+     * Download and extract pool data from the pool CDN, if pool directory does not exist in content folder.
+     *
+     * Will try to reuse existing download if it exists (DownloadHelper will resume download).
+     */
+    public async preloadPoolData() {
+        const poolDirPath = path.join(CONTENT_PATH, "pool");
+        try {
+            await fs.promises.stat(poolDirPath);
+            log.debug("Pool folder exists, skipping download");
+            return;
+        } catch (e) {
+            if ((e as NodeJS.ErrnoException).code !== "ENOENT") {
+                console.error("Error checking pool folder", e);
+                throw e;
+            }
+            // Pool folder does not exist, create and continue
+            await fs.promises.mkdir(poolDirPath);
+        }
+        log.info("Pool folder does not exist, downloading pool data");
+
+        const downloadInfo: DownloadInfo = {
+            type: "game",
+            name: "pool-data",
+            currentBytes: 0,
+            totalBytes: 1,
+        };
+        this.currentDownloads.push(downloadInfo);
+
+        const dlFilePath = path.join(CONTENT_PATH, "data.7z");
+        const dl = new DownloaderHelper(this.poolDataUrl, CONTENT_PATH, {
+            fileName: "data.7z",
+            override: true,
+            resumeIfFileExists: true,
+            resumeOnIncomplete: true,
+            removeOnStop: false,
+            removeOnFail: false,
```

Comment:
Okey, while testing failures, I've notices that no timeouts etc are set here. So, if I go offline, but just disabling my wifi connection, it never fails, it also doesn't fail when I reconnect.

I would recommend taking a look at config in spring-launcher: https://github.com/beyond-all-reason/spring-launcher/blob/2c0a6e17b50ab04ea592c53963c571e5ccb071ba/src/http_downloader.js#L60-L66

Reply:
I forgot to test timeouts/disconnects - will look into it, will come with the data.7z deletion PR.

Should I also set User-Agent to let's say "bar-lobby" while I am at it ?

Reply:
> Should I also set User-Agent to let's say "bar-lobby" while I am at it ?

This one *shouldn't* matter, unless that package starts to be used for some abuse and e.g. Cloudflare will block [default user agent](<https://github.com/hgouveia/node-downloader-helper/blob/e5cc3a5884fb4f90f86b2dae1b918450b6d24d42/example/index.js#L26>) like they do with python urllib. So, yeah, you can set it to bar-lobby, but we aren't consistently setting it across the whole codebase, all http invocations anyway ;).

Reply:
data.7z deletion fix in #336 - I didn't add the user-agent, believe that should be done properly (global constant and changed everywhere) when really needed

Reply:
dammit, I forgot to include the timeout changes